ignite/handlers/checkpoint.py

```diff
@@ -246,18 +261,34 @@ def load_objects(to_load: Mapping, checkpoint: Mapping) -> None:
         Checkpoint._check_objects(to_load, "load_state_dict")
         if not isinstance(checkpoint, collections.Mapping):
             raise TypeError("Argument checkpoint should be a dictionary, but given {}".format(type(checkpoint)))
+
+        def load_state_dict(obj, state_dict):
+            if isinstance(obj, torch.nn.parallel.DataParallel):
+                obj.module.load_state_dict(state_dict)
+            elif isinstance(obj, torch.nn.parallel.DistributedDataParallel):
+                # only load in first process
+                if torch.distributed.get_rank() > 0:
+                    return
+                torch.distributed.barrier()
+                obj.module.load_state_dict(state_dict)
+                obj.require_forward_param_sync = True
```

Comment:
Why do we need these 3 lines ?

Reply:
They are needed, because the state_dict should only be loaded by one process. In opposite to DP in DDP there exist permanent replicas on all devices. These lines tell it to synchronize the loaded parameters before doing another forward to all replicas